torch/autograd/_functions/tensor.py

```diff
@@ -591,4 +591,22 @@ def backward(self, grad_output):
         return grad_input
 
 
+class Cumsum(Function):
+
+    def __init__(self, dim):
+        super(Cumsum, self).__init__()
+        self.dim = dim
+
+    def forward(self, input):
+        return torch.cumsum(input, dim=self.dim)
+
+    def backward(self, grad_output):
+        grad_sum = torch.sum(grad_output, dim=self.dim)
+
+        grad_input = torch.cumsum(-grad_output, dim=self.dim)
+        grad_input += grad_sum.expand_as(grad_input)
```

Comment:
But this would allow a useless summation over the `grad_output` tensor.

What if we:
- select the last element according to self.dim
- clone it so that it doesn't share the memory anymore
- then expand it and subtract it

This would mean removing one summation over the whole tensor, at the cost of a clone of the reduced tensor, which is smaller?